test/scripts/get-missing-cherry-picks.test.ts

```diff
@@ -1,150 +1,222 @@
-          message: `\
-          First PR to fix bug (#29041)
-          
-          * update worker-dom
-          * yarn lock
-          
-          (cherry picked from commit 725595ac41082b0f9db3103143b04070b0fa7a9e)`,
-        },
-      },
-      {
-        commit: {
-          message: `\
-            Second PR to fix bug (#29051)
-            
-            (cherry picked from commit b0fd442d5abc691a3d6eea076eccb83525258b7b)`,
-        },
+    when(
+      spiedRepos.compareCommitsWithBasehead(
+        deepEqual({
+          owner: 'ampproject',
+          repo: 'amphtml',
+          basehead: '2202070001000...2202070001002',
+        })
+      )
+    ).thenResolve({
+      status: 200,
+      data: {
+        commits: [
+          {
+            commit: {
+              message: `\
+              First PR to fix bug (#29041)
+              
+              * update worker-dom
+              * yarn lock
+              
+              (cherry picked from commit 725595ac41082b0f9db3103143b04070b0fa7a9e)`,
+            },
+          },
+          {
+            commit: {
+              message: `\
+              Second PR to fix bug (#29051)
+              
+              (cherry picked from commit b0fd442d5abc691a3d6eea076eccb83525258b7b)`,
+            },
+          },
+        ],
       },
-    ];
-
-    nock(url)
-      .get('/repos/ampproject/amphtml/compare/2202070001000...2202070001002')
-      .reply(200, {commits: cherryPickCommits})
-      .get(
-        '/repos/ampproject/amphtml/compare/2202090001000...725595ac41082b0f9db3103143b04070b0fa7a9e'
+    } as CompareResponse);
+    when(
+      spiedRepos.compareCommitsWithBasehead(
+        deepEqual({
+          owner: 'ampproject',
+          repo: 'amphtml',
+          basehead: '2202090001000...725595ac41082b0f9db3103143b04070b0fa7a9e',
+        })
       )
-      .reply(200, {status: 'diverged'})
-      .get(
-        '/repos/ampproject/amphtml/compare/2202090001000...b0fd442d5abc691a3d6eea076eccb83525258b7b'
+    ).thenResolve({
+      status: 200,
+      data: {status: 'diverged'},
+    } as CompareResponse);
+    when(
+      spiedRepos.compareCommitsWithBasehead(
+        deepEqual({
+          owner: 'ampproject',
+          repo: 'amphtml',
+          basehead: '2202090001000...b0fd442d5abc691a3d6eea076eccb83525258b7b',
+        })
       )
-      .reply(200, {status: 'ahead'});
+    ).thenResolve({
+      status: 200,
+      data: {status: 'ahead'},
+    } as CompareResponse);
 
     const releases = new Set([
       '2202060001000', // lts
       '2202070001002', // stable
       '2202080001000', // beta-traffic
     ]);
-    const commits = await getMissingCommits('2202090001000', releases);
+    const commits = await getMissingCommits(octokit, '2202090001000', releases);
     assert(commits.length == 2);
     assert(
       [
         '725595ac41082b0f9db3103143b04070b0fa7a9e',
         'b0fd442d5abc691a3d6eea076eccb83525258b7b',
       ].every((c) => commits.includes(c))
```

Comment:
Prefer to be as explicit as possible in tests, it's easier to read compared to using container logic

```suggestion
    assert(commits.include('725595ac41082b0f9db3103143b04070b0fa7a9e'));
    assert(commits.include('b0fd442d5abc691a3d6eea076eccb83525258b7b'));
```